### PR TITLE
Label Providers

### PR DESCRIPTION
**Pull request by georgiana-ecobici**
Change-Id: I95f933cfc19c6bdcdd0b1f52f6674859c31c94e4
Signed-off-by: Georgiana Ecobici <georgiana-elena.ecobici@thalesgroup.com>

### CODE REVIEW

File: common/plugins/org.polarsys.capella.common.ui/src/org/polarsys/capella/common/ui/providers/MDEAdapterFactoryLabelProvider.java [hunk 1]
```diff
@@ -98,20 +59,20 @@ public String getText(Object object) {
...
-            (ILabelProviderDelegation) ExtensionPointHelper.createInstance(configurationElements[0], ExtensionPointHelper.ATT_CLASS);
-      }
-      __alreadyLookup = true;
-    }
-    return __delegatedLabelProvider;
-  }
-}
+  protected ILabelProviderDelegation getDelegatedLabelProvider() {
+    if (!__alreadyLookup && (null == __delegatedLabelProvider)) {
+      // Load ITabbedPropertiesLabelProviderDelegation contributor if any.
```
Comment (claudiasima-ro) [inline common/plugins/org.polarsys.capella.common.ui/src/org/polarsys/capella/common/ui/providers/MDEAdapterFactoryLabelProvider.java:66]: Are these comments needed?

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/providers/TitleBlockItemDecoratorAdapterFactory.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+public class TitleBlockItemDecoratorAdapterFactory extends CustomDecoratorAdapterFactory {
+  public TitleBlockItemDecoratorAdapterFactory() {
+    super(new TitleBlockAdapterFactory());
+  }
+
+  @Override
+  protected IItemProviderDecorator createItemProviderDecorator(Object target, Object Type) {
+    if (target instanceof DAnnotation && TitleBlockHelper.isTitleBlockAnnotation((DAnnotation) target)) {
+        return new TitleBlockItemProviderDecorator(this);
```
Comment (claudiasima-ro) [inline core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/providers/TitleBlockItemDecoratorAdapterFactory.java]: There seem to be more than 2 spaces indentation here.

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/providers/TitleBlockItemDecoratorAdapterFactory.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    } 
+    return new ForwardingItemProviderAdapterDecorator(this);
+  }
+  
+  public boolean isFactoryForType(Object type) {
+      return decoratedAdapterFactory.isFactoryForType(type);
```
Comment (claudiasima-ro) [inline core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/providers/TitleBlockItemDecoratorAdapterFactory.java]: There seem to be more than 2 spaces indentation here.

**Review by claudiasima-ro — APPROVED** [review]

File: common/plugins/org.polarsys.capella.common.ui/src/org/polarsys/capella/common/ui/providers/MDEAdapterFactoryLabelProvider.java (hunk 1 above)
Comment (georgiana-ecobici) [reply common/plugins/org.polarsys.capella.common.ui/src/org/polarsys/capella/common/ui/providers/MDEAdapterFactoryLabelProvider.java:66]: Done, formatted all the code.

File: core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/providers/TitleBlockItemDecoratorAdapterFactory.java (hunk 2 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.diagram.helpers/src/org/polarsys/capella/core/diagram/helpers/providers/TitleBlockItemDecoratorAdapterFactory.java]: Done.

**Pull request merged**